### PR TITLE
Move integration tests to release pipeline

### PR DESCRIPTION
**Pull request by slevenick**
Move integration tests to release pipeline

-----------------------------------------------------------------
# [all]

### CODE REVIEW

File: .ci/release.yml.tmpl [hunk 1]
```diff
@@ -67,3 +67,47 @@ jobs:
             params:
               file: dist/terraform-provider-google.{{arch}}
           {% endfor %}
+
+    - name: inspec-integration-test
```
Comment (rambleraptor) [inline .ci/release.yml.tmpl]: Do we want to move this to the release pipeline or to a new `test` pipeline?

Comment (slevenick) [reply .ci/release.yml.tmpl]: Ah yeah, totally spaced on that. I guess that's what we want?

Comment (rambleraptor) [reply .ci/release.yml.tmpl]: I'm pretty sure that's what we want.

Comment (slevenick) [reply .ci/release.yml.tmpl]: IMO that will fragment our pipelines a lot, meaning that we will forget about the IT pipeline or the nightly release pipeline and never check them. But I'm happy to do it

Comment (ndmckinley) [reply .ci/release.yml.tmpl]: I'm inclined just to move it to the release pipeline, but if you have a strong preference for test pipeline, that's okay with me too.

Comment (rambleraptor) [reply .ci/release.yml.tmpl]: Yeah, that's a good point. I have no issues doing a test/release pipeline. Let's just stick with this and we can move towards 3 pipelines if people want to change.

Comment (rambleraptor) [reply .ci/release.yml.tmpl]: I'm convinced! Let's just move to release.

**Pull request merged**